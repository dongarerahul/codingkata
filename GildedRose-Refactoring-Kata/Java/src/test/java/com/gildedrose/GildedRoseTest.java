package com.gildedrose;

import org.junit.jupiter.api.Test;

import static org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations;

class GildedRoseTest {

    @Test
    void foo() {
        verifyAllCombinations(
                this :: doUpdateQuality,
                new String[]  {"foo"},
                new Integer[] { -1  },
                new Integer[] { 0   } );
    }

    private String doUpdateQuality (String name, Integer sellIn, Integer quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
