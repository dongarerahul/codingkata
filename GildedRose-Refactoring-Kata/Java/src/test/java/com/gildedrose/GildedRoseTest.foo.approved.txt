[foo, -1, 0] => foo, -2, 0 
